### DISCUSSION TITLE
How to disable gas-strict-inequalities? (On next line).

### DISCUSSION THREAD

**Discussion by a-t-0** [crytic/slither#2593] [Q&A]
Thank you for your work on providing such an awesome automated tooling to improve the quality of Solidity code!

## Context

When I get the: 
```
  167:15  warning  GC: Non strict inequality found. Try converting to a strict one                                  gas-strict-inequalities
```
warning on line:
```sol
//slither-disable-next-line gas-strict-inequalities
      require(strBytes[i] >= "0" && strBytes[i] <= "9", "String contains non-numeric characters");
```
I tried looking up the `DETECTOR_NAME` accompanying the `gas-strict-inequalities` warning on [this page](https://github.com/crytic/slither/wiki/Detector-Documentation). However I did not yet find the strict inequality warning. So instead I tried what I thought was the closest match:
```sol
//slither-disable-next-line incorrect-equality
      require(strBytes[i] >= "0" && strBytes[i] <= "9", "String contains non-numeric characters");
```

## Question
How can I (find how to) disable the `gas-strict-inequalities` detection on the next line?

## Answer
By looking at SolHint documentation instead of Slither documentation. My apologies for this sloppiness. I did not yet know how to delete this post.

**Comment by adilburaksen** [comment]
Three ways to suppress it:

**1. CLI flag (per run)**
```bash
slither . --exclude gas-strict-inequalities
```

**2. Config file (persistent)**

Add to `.slither.config.json` in your project root:
```json
{
  "exclude_detectors": ["gas-strict-inequalities"]
}
```
Slither picks this up automatically on every run.

**3. Inline suppression (per line)**
```solidity
// slither-disable-next-line gas-strict-inequalities
uint256 result = a >= b ? a : b;
```

Option 2 is usually the cleanest if you want it off project-wide.
